File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        // for (int j=0; j<nums.size(); j++){
        //     numMap[nums[j]] = j;
        // }
        for (int i=0; i < nums.size(); i++){
            int diff = target - nums[i];
                if (numMap.count(diff)){
                    return {i,numMap[diff]};
                
            }else{
                numMap[nums[i]] = i;
            }
        }
    return {0,0};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map1 = new Map();
    for(let i=0; i<nums.length; i++){
        let val = nums[i];
        if (map1.has(target-val)){
            return [map1.get(target-val), i]
        }
        map1.set(val, i);
    }
    return [0,1]
};